llvm/include/llvm/CAS/CASDB.h

```diff
@@ -141,10 +141,11 @@ class CASDB : public CASIDContext {
   virtual CASID getID(ObjectHandle Handle) const = 0;
 
   /// Get a reference to the object called \p ID.
-  ///
-  /// Returns \c None if not stored in this CAS.
   virtual Optional<ObjectRef> getReference(const CASID &ID) const = 0;
 
+  /// Get a reference to the object has the hash value \p Hash.
+  virtual Optional<ObjectRef> getReference(ArrayRef<uint8_t> Hash) const = 0;
```

Comment:
Why do we need this separately from the CASID version?

Reply:
We don't have a method to construct `CASID` from ArrayRef of bytes and ArrayRef is how it is stored in ActionCache.